src/lib.rs

```diff
@@ -146,7 +146,9 @@ impl<T: Clone> MenuState<T> {
     ///
     /// down highlights "sub item 2"
     pub fn down(&mut self) {
-        if self.active_depth() == 1 {
+        if self.active_depth() == 0 {
```

Comment:
should change the code in right?

Reply:
oh right, "right" is going deep, so it won't pop the root_item

Reply:
the change made in "left" should fixed the problem. this change is not required?

Reply:
Oh, I meant to do this in right(), not down().....